awx/main/models/credential/__init__.py

```diff
@@ -540,6 +545,7 @@ def build_extra_vars_file(vars, private_dir):
                 os.chmod(path, stat.S_IRUSR)
                 return path
 
+            extra_vars = render_extravars(self.injectors.get('extra_vars', {}))
```

Comment:
Should this be a call to `build_extra_vars` instead?

Reply:
YES it SHOULD! My bad, I missed that when I refactored my code...